static/src/javascripts/projects/common/modules/commercial/contributions-liveblog-utilities.js

```diff
@@ -0,0 +1,128 @@
+        const blocks = document.getElementsByClassName('block');
+        const blocksToInsertManualEpicAfter = document.getElementsByClassName(
+            INSERT_EPIC_AFTER_CLASS
+        );
+        const epicsAlreadyOnPage = document.getElementsByClassName('is-epic');
+
+        const isLiveblogLongEnoughYet = blocks.length > 4;
+
+        if (
+            blocksToInsertManualEpicAfter.length ||
+            epicsAlreadyOnPage.length ||
+            !isLiveblogLongEnoughYet
+        ) {
+            return [...blocksToInsertManualEpicAfter];
+        }
+
+        const autoBlockNum = Math.floor(Math.random() * 3) + 1;
+        const blockToInsertAutoEpicAfter = blocks[autoBlockNum];
+
+        return [...blocksToInsertManualEpicAfter].concat(
+            blockToInsertAutoEpicAfter
+        );
+    });
+
+const setEpicLiveblogEntryTimeData = (
+    el: Element,
+    timeData: TimeData
+): void => {
+    const epicTimeEl = el.querySelector('time');
+    const epicAbsoluteTimeEl = el.querySelector('.block-time__absolute');
+
+    if (epicTimeEl && epicAbsoluteTimeEl) {
+        const epicTimeLink = epicTimeEl.parentNode;
+        if (epicTimeLink instanceof HTMLAnchorElement) {
+            epicTimeLink.href = timeData.blockHref;
+        }
+
+        epicTimeEl.setAttribute('datetime', timeData.datetime);
+        epicTimeEl.setAttribute('title', timeData.title);
+        epicTimeEl.innerHTML = timeData.date;
+        epicAbsoluteTimeEl.innerHTML = timeData.time;
+    }
+};
+
+const setupViewTracking = (el: HTMLElement, test: EpicABTest): void => {
+    // top offset of 18 ensures view only counts when half of element is on screen
+    const inView = elementInView(el, window, {
+        top: 18,
+    });
+
+    inView.on('firstview', () => {
+        logView(test.id);
+        mediator.emit(test.viewEvent);
+    });
+};
+
+const addEpicToBlocks = (epicHtml: string, test: EpicABTest): Promise<void> =>
+    getBlocksToInsertEpicAfter().then(blocksToInsertEpicAfter => {
+        blocksToInsertEpicAfter.forEach(el => {
+            getLiveblogEntryTimeData(el).then((timeData: TimeData) => {
+                fastdom.write(() => {
+                    const $epic = $.create(epicHtml);
+                    $epic.insertAfter(el);
+                    mediator.emit(test.insertEvent);
+                    $(el).removeClass(INSERT_EPIC_AFTER_CLASS);
+                    setEpicLiveblogEntryTimeData($epic[0], timeData);
+                    setupViewTracking(el, test);
+                });
+            });
+        });
+    });
+
+export const setupEpicInLiveblog = (
+    epicHtml: string,
+    test: EpicABTest
+): void => {
+    addEpicToBlocks(epicHtml, test);
+
+    if (!isAutoUpdateHandlerBound) {
+        mediator.on('modules:autoupdate:updates', () => {
+            addEpicToBlocks(epicHtml, test);
+        });
+        isAutoUpdateHandlerBound = true;
+    }
+};
+
+export const pageId: string = config.get('page.pageId', '');
```

Comment:
seems a bit weird to export this? I'd just call `config.get('page.pageId', '')` wherever you need it

Reply:
*actually...* why don't we just get rid of it and that `campaignSuffix` nonsense? it's a relic of a time before `referrerUrl` was in the data lake - nowadays it just adds noise to campaign codes (should have been deleted a long time ago in fact!)

you could set campaignCode to `world_cup_2018` for your liveblog test if that's easy, could make later analysis easier by allowing easy grouping along with email, social, etc - though I suspect due to componentId/campaignCode confusion and legacy code it might be a pain. if so don't bother